docs/interfaces.md

```diff
@@ -0,0 +1,431 @@
+```go
+type EnclaveRegistryCC struct {
+	ra  attestation.Verifier
+}
+
+// Intel IAS verification key (same as used by TLCC Enclave)
+const IntelPubPEM = `...`
+
+type IASAttestationReport struct {
+	EnclavePk                   []byte `json:"EnclavePk"`
+	IASReportSignature          string `json:"IASReport-Signature"`
+	IASReportSigningCertificate string `json:"IASReport-Signing-Certificate"`
+	IASReportBody               []byte `json:"IASResponseBody"`
+}
+```
+
+
+# TLCC
+
+## Interface:
+The TLCC interface is implemented through a "normal" chaincode interface.
+
+```go
+// chaincode interface 
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// TODO note that we are multiplexing normal and range queries with this method
+// used by ecc via cc2cc
+getMetadata(key string, nonce string, isRangeQuery bool) (cmacBase64  string, error)
+
+validateMSP(mspId string, nonce string) (valid bool, cmacBase64 string, error)
+
+// used by the admin to join the channel
+joinChannel(channel_id string) error
+```
+
+```c++
+// interface exposed to TLCC_enclave via cgo
+// logging
+void ocall_print_string([in, string] const char *str);
+```
+
+## State:
+TLCC keeps a local reference of the enclave id. This reference is volatile. Also a reference to the peer is kept to access the ledger and read blocks.
+
+```go
+type TrustedLedgerCC struct {
+	enclave enclave.Stub
+	peer    *peer.Peer
+}
+
+type StubImpl struct {
+	eid C.enclave_id_t
+}
+```
+
+
+# TLCC_Enclave
+
+## Interface:
+Defined in the TLCC EDL.
+```c++
+// TBD is this interface already sufficient to support multi-channel?
+// TODO check if a block contatains any channel information?
+
+// creates a new channel
+public int ecall_join_channel(
+        [in, size=gen_len] uint8_t *genesis, uint32_t gen_len);
+
+// pushes next block to tlcc
+public int ecall_next_block(
+        [user_check] uint8_t *block_bytes, uint32_t block_size);
+
+// verify state data
+public int ecall_get_state_metadata(
+        [in, string] const char *key, // TBD: should we add namespace separetely?
+        [in, size=32] uint8_t *nonce,
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+public int ecall_get_multi_state_metadata(
+        [in, string] const char *comp_key,
+        [in, size=32] uint8_t *nonce,
+        [out] sgx_cmac_128bit_tag_t *cmac);
+
+// verify msp identities
+public int ecall_validate_msp_id(
+        [in, string] const char *msp_id,
+        [in, size=32] uint8_t *nonce,
+        [out] bool isValid, 
+        [out] sgx_cmac_128bit_tag_t *cmac);
+```
+
+## State:
+The internal TLCC state consists of the current ledger state (block number, integrity-metadata, and msp) and sessions information for ecc communication.
+
+```c++
+// TODO: this needs some more brain cycles with respect of #402 and #410
+typedef struct internal_tlcc_state_t {
+        const char* INTEL_PUB_PEM, // Intel IAS verification key
+        
```

Comment:
This should probably be outsourced to the attestation module?  Practically, i thought this also would really be some static variable/constant defined in that code and so wouldn't really need any runtime setting, other than the other data you have here. So keeping that separate seems anyway kindof nice ...